src/utils/plrStakingHelper.tsx

```diff
@@ -0,0 +1,345 @@
+let rewardToken: string = null;
+let storedContractInfo: IStakingContractInfo = null;
+
+export const mapWalletTypeToIcon = (type: WalletType) => {
+  switch (type) {
+    case WalletType.ETHERSPOT:
+      return 'etherspot16';
+    case WalletType.KEYBASED:
+      return 'wallet16';
+    case WalletType.ARCHANOVA:
+      return 'pillar16';
+    default:
+      return 'etherspot16';
+  }
+};
+
+export const mapWalletTypeToName = (type: WalletType) => {
+  switch (type) {
+    case WalletType.ETHERSPOT:
+      return 'Etherspot';
+    case WalletType.KEYBASED:
+      return 'Keybased';
+    case WalletType.ARCHANOVA:
+      return 'Pillar v1';
+    default:
+      return 'Etherspot';
+  }
+};
+
+export const showBalancesReceivedToast = (symbol: string, amount: BigNumber) => {
+  Toast.show({
+    message: `Received ${ethers.utils.formatEther(amount)} ${symbol}! Staking PLR now...`,
+    emoji: 'tada',
+    autoClose: true,
+  });
+};
+
+export const showStakedTokensReceivedToast = (symbol: string, amount: BigNumber) => {
+  Toast.show({
+    message: `${ethers.utils.formatEther(amount)} PLR has been staked! Received ${ethers.utils.formatEther(
+      amount,
+    )} ${symbol}!`,
+    emoji: 'tada',
+    autoClose: false,
+  });
+};
+
+export const getStakingContractAddress = () => {
+  const stakingContract = remoteConfig.getString(REMOTE_CONFIG.PLR_STAKING_CONTRACT);
+  return stakingContract;
+};
+
+export const validatePlrStakeAmount = (plrAmount: BigNumber) => {
+  if (!plrAmount) return false;
+
+  const bnAmount = ethers.utils.parseUnits(plrAmount.toString());
+
+  if (bnAmount.gte(minStakingAmount) && bnAmount.lte(maxStakingAmount)) return true;
+
+  return false;
+};
+
+export const getStakingRemoteConfig = (): IStakingRemoteConfig => {
+  const stakingContract = remoteConfig.getString(REMOTE_CONFIG.PLR_STAKING_CONTRACT);
+  const stakedToken = remoteConfig.getString(REMOTE_CONFIG.PLE_STAKING_TOKEN_ADDRESS);
+  const featureStaking = remoteConfig.getBoolean(REMOTE_CONFIG.FEATURE_PLR_STAKING);
+  const stakingStartTime = remoteConfig.getNumber(REMOTE_CONFIG.PLR_STAKING_START_TIME);
+  const stakingLockedStartTime = remoteConfig.getNumber(REMOTE_CONFIG.PLR_STAKING_LOCKED_START_TIME);
+
+  return { stakingContract, stakedToken, featureStaking, stakingStartTime, stakingLockedStartTime };
+};
+
+export const getStakingContract = () => {
+  const sdk: Sdk = etherspotService.getSdkForChain(CHAIN.ETHEREUM);
+
+  if (!sdk) return null;
+
+  if (!stakingContract) {
+    const contractAddress = getStakingContractAddress();
+    stakingContract = sdk.registerContract<IStakingContractAbi>('PStaking', plrStakingAbi, contractAddress);
+  }
+
+  return stakingContract;
+};
+
+export const getStakingContractState = async () => {
+  if (!stakingContractState) {
+    const stakingContract = getStakingContract();
+    if (!stakingContract) return null;
+
+    stakingContractState = await stakingContract.callGetContractState();
```

Comment:
What happens if this call fails? Just thinking about how fragile blockchain calls can be...